src/Integration/Exclusions/ExclusionSettingsFileStorage.cs

```diff
@@ -18,22 +18,80 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using System;
 using System.ComponentModel.Composition;
+using System.IO;
+using System.IO.Abstractions;
+using Newtonsoft.Json;
 using SonarLint.VisualStudio.Core.Analysis;
+using SonarLint.VisualStudio.Core.Binding;
+using SonarLint.VisualStudio.Integration.Helpers;
+using SonarLint.VisualStudio.Integration.Resources;
 using SonarQube.Client.Models;
+using static System.String;
 
 namespace SonarLint.VisualStudio.Integration.Exclusions
 {
     [Export(typeof(IExclusionSettingsFileStorage))]
     internal class ExclusionSettingsFileStorage : IExclusionSettingsFileStorage
     {
+        private readonly IConfigurationProvider bindingConfigProvider;
+        private readonly ILogger logger;
+        private readonly IFileSystem fileSystem;
+
+        [ImportingConstructor]
+        public ExclusionSettingsFileStorage(IConfigurationProvider bindingConfigProvider, ILogger logger)
+            : this(bindingConfigProvider, logger, new FileSystem())
+        {
+        }
+
+        internal ExclusionSettingsFileStorage(IConfigurationProvider bindingConfigProvider, 
+            ILogger logger,
+            IFileSystem fileSystem)
+        {
+            this.bindingConfigProvider = bindingConfigProvider;
+            this.logger = logger;
+            this.fileSystem = fileSystem;
+        }
+
         public void SaveSettings(ServerExclusions settings)
         {
         }
 
         public ServerExclusions GetSettings()
         {
+            try
+            {
+                var bindingConfiguration = bindingConfigProvider.GetConfiguration();
+
+                if (bindingConfiguration.Mode == SonarLintMode.Standalone)
```

Comment:
I ended up leaving the standalone check. The AnalyzableIndicator shouldn't do this check, so it's ok that it's here.

Reply:
I still think a `Debug.Fail` would be useful in this case so that we have a chance of detecting the issue during development.

Reply:
During Read it won't be a mistake, it's a real use case.
AnalysisController calls AnalyzableIndicator, that calls this class. None of them know anything about connected mode.  So they would call this even when they're in standalone.